src/shared_realm.cpp

```diff
@@ -300,14 +300,13 @@ void Realm::update_schema(Schema schema, uint64_t version, MigrationFunction mig
                 return reset_file_if_needed(schema, version, required_changes);
 
             case SchemaMode::Additive:
-                if (required_changes.empty()) {
-                    if (version == m_schema_version) {
-                        set_schema(std::move(schema), version);
-                        return true;
-                    }
+                if (ObjectStore::verify_valid_additive_changes(required_changes, version > m_schema_version))
```

Comment:
It might be clearer to hoist `version > m_schema_version` into a variable so it can be given a name. It's not obvious without looking at the parameter name what this would mean.